### ISSUE TITLE
Nested `send -K` invocations sometimes execute in the wrong order

### ISSUE THREAD

**Issue by Hexstream** [tmux/tmux#3476]
* tmux version (`tmux -V`): `tmux next-3.4` (built from latest commit `b55f34029ac05474dfd993c187b9c61bbcd4e1a1` from 10 february 2023 with no patches applied)
* Platform (`uname -sp`): `Linux unknown` (Debian 11)
* $TERM inside and outside of tmux (`echo $TERM`): `tmux-256color` inside, `xterm-256color` outside.
* Logs from tmux (`tmux kill-server; tmux -vv new`): Please use provided `menu.conf`.

Hello,

This is a dedicated issue for [the bug I previously reported](https://github.com/tmux/tmux/issues/3361#issuecomment-1397647084), but with a better `menu.conf` which sends characters instead of opening windows and is designed for easier debugging using the log, although it is still difficult.

Please try this `menu.conf` with `tmux -f menu.conf`:

```
bind -n C-M-a switch-client -T command

bind m menu -T 'Test' -x P -y P -S - \
'Test a (1 2 3)' a {
      send -K C-M-a a
} \
'Test b (4 5 6)' b {
      send -K C-M-a b
} \
'Test c (7 8 9)' c {
      send -K C-M-a c
} \
'Test d ((1 2 3) (4 5 6))' d {
      send -K C-M-a d
} \
'Test e ((1 2 3) (4 5 6) (7 8 9))' e {
      send -K C-M-a e
} \
'Test f (((1 2 3) (4 5 6)) (7 8 9))' f {
      send -K C-M-a f
}

bind -T command a {
     send 'AAAAA '
     send 'ONE '
     send 'TWO '
     send 'THREE '
}

bind -T command b {
     send 'BBBBB '
     send 'FOUR '
     send 'FIVE '
     send 'SIX '
}

bind -T command c {
     send 'CCCCC '
     send 'SEVEN '
     send 'EIGHT '
     send 'NINE '
}

bind -T command d {
     send 'DDDDD '
     send -K C-M-a a
     send -K C-M-a b
}

bind -T command e {
     send 'EEEEE '
     send -K C-M-a a
     send -K C-M-a b
     send -K C-M-a c
}

bind -T command f {
     send 'FFFFF '
     send -K C-M-a d
     send -K C-M-a c
}
```

All tests work as expected, except the last one:

`Test f` (C-b m f) yields
`FFFFF DDDDD CCCCC SEVEN EIGHT NINE AAAAA ONE TWO THREE BBBBB FOUR FIVE SIX ` rather than
`FFFFF DDDDD AAAAA ONE TWO THREE BBBBB FOUR FIVE SIX CCCCC SEVEN EIGHT NINE `.

**Comment by nicm** [comment]
I don't know if you still care about this but if so can you please try this against `master`: [send_K.diff.txt](https://github.com/user-attachments/files/28762916/send_K.diff.txt)

**Comment by Hexstream** [comment]
_Absolutely!!_ As I recall, working around this bug complicated my advanced tmux config quite a bit!

I'm extremely excited to try this patch soon!

**Comment by Hexstream** [comment]
**KICKASS!!! _IT WORKS!!!_** 🎉🎉🎉 **_THANK YOU!!!_** ❤️❤️❤️

This was _100% painless,_ I just trivially updated to latest master dc22196394d38098500b9f68d175ac596f54f940 (I hadn't updated in years), loaded my advanced tmux config, no issues, then I tried your patch, no issues with menu.conf or my config, then I updated my config to remove the workarounds, no issues as well!

I'll publish the diff for my config once your patch lands in master. :)

_I'm **very excited** to see tmux development is still going strong!_ 💪

(**edit:** There is one minor unrelated issue after all, but I'll open another issue once your patch lands in master...)

**Comment by nicm** [comment]
Great, I have applied this to OpenBSD now (with a couple of other minor changes), will be in GitHub later. Thanks!

**Comment by Hexstream** [comment]
Thank you, I was able to [simplify my tmux config](https://github.com/Hexstream/status-quo.hexstream.expert/commit/03ed04b2e2ad8318c8bf551ae3da5c3210bb2ad4)! 👍

**Comment by Hexstream** [comment]
(The minor unrelated issue was [an easy fix](https://github.com/Hexstream/status-quo.hexstream.expert/commit/7ee83268f9559c630e3ebb70087c808452075246) after all...)
